### PR TITLE
PC-1122: Investigation into updating C# & .NET versions

### PR DESCRIPTION
**Pull request by Glenn-Clarke**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1122)

# Description

- Updating C# from 10 to 11
- Updating .Net from 6 to 8
- Changing all references from old versions to new versions
- Breaking change in .Net 8 requires references to port 80 to become port 8080.
- Updated SDK version in global.json from 5.0 -> 8.0
- Some additional warnings in editor (115 warnings to 150)

# Checklist (Investigation results, not full PR)

- [ ] I have made any necessary updates to the documentation
- [ ] I have checked there are no unnecessary IDE warnings in this PR
- [ ] I have checked there are no unintentional line ending changes
- [ ] I have added tests where applicable
- [ ] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [ ] If I have made any changes to website flow, I have checked forward and back behaviour is still consistent
- [ ] If I have made any changes to content strings or resource files, I have followed [the documentation](https://github.com/UKGovernmentBEIS/beis-simple-energy-advice-beta/blob/main/docs/LocalisationDocumentation.md)

### CODE REVIEW

File: SeaPublicWebsite.Data/SeaPublicWebsite.Data.csproj [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
         <ImplicitUsings>enable</ImplicitUsings>
+        <LangVersion>11</LangVersion>
```
Comment (jdgage) [inline SeaPublicWebsite.Data/SeaPublicWebsite.Data.csproj:6]:
```suggestion
        <LangVersion>12</LangVersion>
```

File: SeaPublicWebsite.BusinessLogic/SeaPublicWebsite.BusinessLogic.csproj [hunk 2]
```diff
@@ -1,8 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
         <ImplicitUsings>enable</ImplicitUsings>
+        <LangVersion>11</LangVersion>
```
Comment (jdgage) [inline SeaPublicWebsite.BusinessLogic/SeaPublicWebsite.BusinessLogic.csproj:6]: Expecting this to be C# 12
```suggestion
        <LangVersion>12</LangVersion>
```

File: SeaPublicWebsite.UnitTests/SeaPublicWebsite.UnitTests.csproj [hunk 3]
```diff
@@ -1,10 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
         <IsPackable>false</IsPackable>
         <RootNamespace>Tests</RootNamespace>
         <IsPublishable>false</IsPublishable> 
+        <LangVersion>11</LangVersion>
```
Comment (jdgage) [inline SeaPublicWebsite.UnitTests/SeaPublicWebsite.UnitTests.csproj:8]:
```suggestion
        <LangVersion>12</LangVersion> 
```

File: SeaPublicWebsite/SeaPublicWebsite.csproj [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
     <PropertyGroup>
-        <TargetFramework>net6.0</TargetFramework>
-        <LangVersion>10</LangVersion>
+        <TargetFramework>net8.0</TargetFramework>
+        <LangVersion>11</LangVersion>
```
Comment (jdgage) [inline SeaPublicWebsite/SeaPublicWebsite.csproj:5]:
```suggestion
        <LangVersion>12</LangVersion>
```

File: global.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "5.0",
```
Comment (jdgage) [inline global.json:3]: Puzzling that this was on 5.0 before - was it definitely supposed to be referring to .NET version?

**Review by jdgage — CHANGES_REQUESTED** [review]
Once C# has been bumped up again, I think we want to get this merged to dev and given a good test.

File: global.json (hunk 5 above)
Comment (Glenn-Clarke) [reply global.json:3]: Ah, this seems to actually be the version that .Net CLI uses. I'll change it back

https://learn.microsoft.com/en-us/dotnet/core/tools/global-json

Comment (jdgage) [reply global.json]: I actually think we probably should update this tbh. Suspect it was missed when updating to .NET 6. Sorry fort the back-and-forth!

Comment (Glenn-Clarke) [reply global.json]: No worries, I'll update it now :)

**Review by jdgage — DISMISSED** [review]
Cool, all good for when we pick this up again

**Comment by Glenn-Clarke** [comment]
Had to resolve conflicts with the `build-and-run-tests.yml` so I've rebased onto the latest dev.

**Review by jdgage — APPROVED** [review]

**Pull request merged**